src/loader/EptLoader.js

```diff
@@ -0,0 +1,322 @@
+
+    if (b)	min.y += dst.y / 2;
+    else	max.y -= dst.y / 2;
+
+    if (c)	min.z += dst.z / 2;
+    else	max.z -= dst.z / 2;
+
+    return new Potree.EptKey(
+        this.ept,
+        new THREE.Box3(min, max),
+        this.d + 1,
+        this.x * 2 + a,
+        this.y * 2 + b,
+        this.z * 2 + c);
+  }
+
+  children() {
+    var result = [];
+    for (var a = 0; a < 2; ++a) {
+      for (var b = 0; b < 2; ++b) {
+        for (var c = 0; c < 2; ++c) {
+          var add = this.step(a, b, c).name();
+          if (!result.includes(add)) result = result.concat(add);
+        }
+      }
+    }
+    return result;
+  }
+}
+
+Potree.PointCloudEptGeometryNode = class PointCloudEptGeometryNode extends Potree.PointCloudTreeNode {
+  constructor(ept, b, d, x, y, z) {
+    super();
+
+    this.ept = ept;
+    this.key = new Potree.EptKey(
+        this.ept,
+        b || this.ept.boundingBox,
+        d || 0,
+        x,
+        y,
+        z);
+
+    this.id = PointCloudEptGeometryNode.IDCount++;
+    this.geometry = null;
+    this.boundingBox = this.key.b;
+    this.tightBoundingBox = this.boundingBox;
+    this.spacing = this.ept.spacing / Math.pow(2, this.key.d);
+    this.boundingSphere = U.sphereFrom(this.boundingBox);
+
+    // These are set during hierarchy loading.
+    this.hasChildren = false;
+    this.children = { };
+    this.numPoints = -1;
+
+    this.level = this.key.d;
+    this.loaded = false;
+    this.loading = false;
+    this.oneTimeDisposeHandlers = [];
+
+    let k = this.key;
+    this.name = this.toPotreeName(k.d, k.x, k.y, k.z);
+    this.index = parseInt(this.name.charAt(this.name.length - 1));
+  }
+
+  isGeometryNode() { return true; }
+  getLevel() { return this.level; }
+  isTreeNode() { return false; }
+  isLoaded() { return this.loaded; }
+  getBoundingSphere() { return this.boundingSphere; }
+  getBoundingBox() { return this.boundingBox; }
+  url() { return this.ept.url + 'ept-data/' + this.filename(); }
+  getNumPoints() { return this.numPoints; }
+
+  filename() { return this.key.name(); }
+
+  getChildren() {
+    let children = [];
+
+    for (let i = 0; i < 8; i++) {
+      if (this.children[i]) {
+        children.push(this.children[i]);
+      }
+    }
+
+    return children;
+  }
+
+  addChild(child) {
+    this.children[child.index] = child;
+    child.parent = this;
+  }
+
+  load() {
+    if (this.loaded || this.loading) return;
+    if (Potree.numNodesLoading >= Potree.maxNodesLoading) return;
+
+    this.loading = true;
+    ++Potree.numNodesLoading;
+
+    if (this.numPoints == -1) this.loadHierarchy();
+    this.loadPoints();
+  }
+
+  loadPoints(){
+    this.ept.loader.load(this);
+  }
+
+  async loadHierarchy() {
+    let nodes = { };
+    nodes[this.filename()] = this;
+    this.hasChildren = false;
+
+    let eptHierarchyFile =
+      `${this.ept.url}ept-hierarchy/${this.filename()}.json`;
+
+    const sign = Potree.getSignatureKeyForPath(`${this.filename()}.json`);
```

Comment:
Shouldn’t you sign the eptHierearchyFile address instead?

Reply:
I cannot because of the URL that is added at the beginning of **eptHierarchyFile**. Anyway, I could probably prepend a file name with  `ept-hierarchy/` but I cannot test that right now. I will come back to this as soon as its possible.

Reply:
@nicolas-chaulet I updated that so right now I am looking for a file with `ept-hierarchy` prefix.